micro-rdk/src/esp32/pin.rs

```diff
@@ -185,34 +185,26 @@ impl Esp32GPIOPin {
         self.interrupt_type.is_some()
     }
 
-    pub fn setup_interrupt(&mut self, intr_type: InterruptType) -> Result<(), BoardError> {
-        match &self.interrupt_type {
-            Some(existing_type) => {
-                if *existing_type == intr_type {
-                    return Ok(());
-                }
-            }
-            None => {
-                self.interrupt_type = Some(intr_type);
-            }
-        };
+    pub(crate) fn setup_interrupt(
+        &mut self,
+        intr_type: InterruptType,
+        cb: gpio_isr_t,
+        arg: *mut core::ffi::c_void,
+    ) -> Result<(), BoardError> {
+        self.interrupt_type = Some(intr_type);
```

Comment:
regardless of what the previous interrupt type was, we still set it. before, we just returned `Ok` if the interrupt type was the same, but we wouldn't be able to handle if a module wanted to register a different callback that uses the same interrupt type as was originally set during board setup.